tools/benchmark/src/Reporter.ts

```diff
@@ -193,8 +186,12 @@ export class BenchmarkReporter {
 			if (isResultError(value)) {
 				countFailure++;
 			} else {
-				benchmarkPeriodsSeconds.push(value.stats.arithmeticMean);
-				sumRuntime += value.elapsedSeconds;
+				if ((value as BenchmarkData).customData["Heap Used Avg"] !== undefined) {
+					benchmarkPeriodsSeconds.push(
+						(value as BenchmarkData).customData["Heap Used Avg"] as number,
+					);
+				}
+				sumRuntime += (value as BenchmarkData).elapsedSeconds;
```

Comment:
The casts here shouldn't be necessary, in this block `isResultError()` returned false, so `value` is not a `BenchmarkError` and thus can only be a `BenchmarkData`.

Also, we have a bug here, "Heap Used Avg" is a key that only exists in memory tests, but this file applies to any kind of benchmark. "Heap Used Avg" also isn't related to the running time of the benchmarks, which is what `benchmarkPeriodSeconds` is supposed to be.

@CraigMacomber any thoughts on removing benchmarkPeriodSeconds from here, and lose the "geometric mean (ns)" column from the summary table? It feels a bit specific to the runtime tests... maybe memory tests could benefit from it just to know how much time the iterations are taking, but for custom benchmarks that will only be run once it seems unnecessary. Also, not sure how useful it is at the suite level that the summary table outputs...? A suite could have a mix of very quick and very lengthy benchmarks inside it, and I'm not clear what we gain from having the geometric mean of the durations of individual benchmarks reported.

Reply:
I'm fine with losing the geometric mean. We can always add it back by providing some way to annotate columns for a summary policy or something, and this refactoring is more important than that feature in my opinion.

As for benchmarkPeriodSeconds, Thats new in this PR and I'm not sure what it is. It sounds like its in seconds, so adding some number of bytes to it seems wrong. If its supposed to measure how long the benchmark ran for, that could be useful, but if so it should measure that for both memory and time tests, and probably be names something like total elapsed runtime to make that clear. If its just an accumulates or some arbitrary scores, it needs a better name.

Reply:
@chentong7 let's get rid of benchmarkPeriodSeconds and the geometric mean column.

Reply:
On it!

Reply:
Proposed type test change: https://github.com/microsoft/FluidFramework/pull/21876 (discussion can move over there)